lib/api/src/js/instance.rs

```diff
@@ -19,8 +23,10 @@ use std::fmt;
 #[derive(Clone)]
 pub struct Instance {
     _handle: StoreHandle<WebAssembly::Instance>,
+    thread_seed: Arc<AtomicU32>,
     module: Module,
-    #[allow(dead_code)]
+    /// Reactors are used to wake up the program
+    pub reactors: Reactors,
```

Comment:
Shouldn't the programs use SharedArrayBuffer to wake up? (I might be confusing things, so it would be good to have a bit of better details on for which cases `Reactors` are used)

Reply:
That would make the scope of a SharedArrayBuffer functionality wider then...well shared buffers. We should try and follow the "separation of concerns" principle when we can. Do you think we need to create a superset object/class? and if so what would you name it?

The reactors are needed because WASIX provides reactors (code execution stops until work arrives to be done) - this is essential for building asynchronous programs (for instance with Tokio). This will also be important for low overhead density of web assembly applications running on the same physical server, if there are web assembly programs running that are just in a spin lock / futex waiting on syscall to finish (e.g. poll) then it will not be possible to push them into a warm state (loaded but no CPU cycles)

Reply:
I moved this away from the instance and put it in Wasi instead - its much cleaner now